crypto/asn1/a_mbstr.c

```diff
@@ -272,12 +270,15 @@ static int out_utf8(unsigned long value, void *arg)
 static int type_str(unsigned long value, void *arg)
 {
     unsigned long types;
+    const int native = ossl_fromascii(value);
```

Comment:
Let's not overcomplicate it :-) It does work in all *practical* cases. Simplest would probably be to "saturate" it, e.g. `native = value > INT_MAX ? INT_MAX : ossl_fromascii(value)`